esp/esp/program/modules/base.py

```diff
@@ -212,6 +213,8 @@ def findModule(request, tl, one, two, call_txt, extra, prog):
                 other_modules = moduleobj.findCategoryModules(False)
                 for m in other_modules:
                     m.request = request
+                    if request.user.is_morphed() and not isinstance(m, RegProfileModule):
```

Comment:
Maybe this should be `request.user.updateOnsite(request)`? (This is the call that is used to determine whether to skip deadlines when you're using the onsite morph, as opposed to normal morph.)